### PR TITLE
Add Gradle Develocity

### PR DESCRIPTION
**Pull request by lrytz**
This enables develocity for Jenkins builds (PR and merge validation). We could also/alternatively enable it in GitHub actions.

Scala 2 version of https://github.com/scala/scala3/pull/21386.

### CODE REVIEW

**Comment by lrytz** [comment]
```
[info] Publishing build scan...
[info] Publishing failed.
[info] The response from https://develocity.scala-lang.org/scans/publish/sbt/1.1/token was not from Develocity.
[info] The specified server address may be incorrect, or your network environment may be interfering.
[info] Please report this problem to your Develocity administrator via https://develocity.scala-lang.org/help and include the following via copy/paste:
[info] ----------
[info] Sbt version: 1.10.1
[info] Plugin version: 1.1
[info] Request URL: [https://develocity.scala-lang.org/scans/publish/sbt/1.1/token](https://develocity.scala-lang.org/scans/publish/sbt/1.1/token%1B[0m)
[info] Request ID: 437c7aca-5d0a-4b08-9e8b-0c8a4b6d717b
[info] Response status code: 403
[info] Response content type: text/html; charset=UTF-8
[info] Response server type: cloudflare
[info] ----------
```

**Comment by lrytz** [comment]
maybe i need to set up a firewall rule on aws

**Comment by lrytz** [comment]
The issue reproduces locally on Java 8, publishing works on Java 11+

**Comment by som-snytt** [comment]
I can already feel the wind in my hair.

**Comment by lrytz** [comment]
https://develocity.scala-lang.org/scans?search.rootProjectNames=scala2

File: build.sbt [hunk 1]
```diff
@@ -93,6 +93,53 @@ ThisBuild / headerLicense  := Some(HeaderLicense.Custom(
      |""".stripMargin
 ))
 
+ThisBuild / develocityConfiguration := {
+  def pubDev = Option(System.getProperty("scala.build.publishDevelocity"))
+  val isInsideCI = sys.env.get("JENKINS_URL").exists(_.contains("scala-ci.typesafe.com"))
+  val previousConfig = develocityConfiguration.value
+  val previousBuildScan = previousConfig.buildScan
+  val previousBuildCache = previousConfig.buildCache
+  val isBuildCacheEnabled = sys.BooleanProp.keyExists("build-cache.enabled") // same prop in Scala 3
```
Comment (adpi2) [inline build.sbt]: We don't have it in Scala 3, and I don't know if I am ever going to set it up.

File: build.sbt [hunk 2]
```diff
@@ -93,6 +93,53 @@ ThisBuild / headerLicense  := Some(HeaderLicense.Custom(
      |""".stripMargin
 ))
 
+ThisBuild / develocityConfiguration := {
+  def pubDev = Option(System.getProperty("scala.build.publishDevelocity"))
```
Comment (adpi2) [inline build.sbt:99]: Not really clear what values this property can contain.

File: scripts/jobs/validate/publish-core [hunk 3]
```diff
@@ -28,7 +28,7 @@ case $prDryRun in
     if $libraryAvailable && $reflectAvailable && $compilerAvailable; then
       echo "Scala core already built!"
     else
-      $SBT -warn "setupPublishCore $prRepoUrl" publish
+      $SBT -Dscala.build.publishDevelocity=build -warn "setupPublishCore $prRepoUrl" publish
```
Comment (adpi2) [inline scripts/jobs/validate/publish-core:31]: Why not using the name of the script: `publish-core` instead of `build`?

**Review by adpi2 — APPROVED** [review]
Some minor comments, but overall it looks good to me.

It seems you are cautious on what gets published, with `-D:scala.build.publishDevelocity`. What's the reasoning behind that?

**Comment by lrytz** [comment]
> It seems you are cautious on what gets published, with `-D:scala.build.publishDevelocity`. What's the reasoning behind that?

The build invokes `sbt` to generate some properties file, having scans for that seems just unnecessary noise.

File: build.sbt (hunk 1 above)
Comment (lrytz) [reply build.sbt]: Ah, I looked at a different commit. Will align with Scala 3.

File: build.sbt (hunk 2 above)
Comment (lrytz) [reply build.sbt:99]: Right, I'll add a comment.

File: scripts/jobs/validate/publish-core (hunk 3 above)
Comment (lrytz) [reply scripts/jobs/validate/publish-core:31]: The value is turned into the `JENKINS_STAGE` value on build scans, eg https://develocity.scala-lang.org/s/gqtejqozpexas/custom-values, `build` and `test` just looked nice 🤷‍♂️

**Comment by SethTisue** [comment]
(I canceled the Travis-CI build here so the 2.12.20 publishing job can run.)

**Comment by SethTisue** [comment]
consider targeting 2.12.x branch rather than 2.13.x? (for later forward-merge to 2.13.x, of course)

**Comment by lrytz** [comment]
2.12: https://github.com/scala/scala/pull/10866

**Pull request closed without merging**